### PR TITLE
Adds WSGI and Architecture Docs

### PR DESCRIPTION
**Pull request by bmbouter**
This creates an Architecture and Scaling top level section of the docs.
It also explains for each component how it can be scaled which mostly
involves documenting how the WSGI application is laid out.

https://pulp.plan.io/issues/3132
closes #3132

### CODE REVIEW

File: docs/arch_and_scale/index.rst [hunk 1]
```diff
@@ -0,0 +1,33 @@
+.. _RabbitMQ Clustering Guide: https://www.rabbitmq.com/clustering.html
+
+Architecture and Scaling
+========================
+
+Pulp's architecture has five components of it. Each of these can be horizontally scaled
```
Comment (CodeHeeler) [inline docs/arch_and_scale/index.rst]: Minor point: "has five components." or "has five components to it" I think sounds better than of it here.

File: docs/arch_and_scale/index.rst [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+WSGI application
+  Pulp's web application is served by one or more WSGI webservers. See the
+  :ref:`wsgi-application` docs for more info on deploying and scaling this component.
+
+Workers
+  Pulp's tasking system requires at least one running worker. Additional workers can be added to add
+  capacity to the tasking system.
+
+Resource Manager
+  Pulp's tasking system requires at least one running resource manager. Although its required for
```
Comment (CodeHeeler) [inline docs/arch_and_scale/index.rst]: "Although it's"--missing the apostrophe in the contraction.

File: docs/arch_and_scale/index.rst (hunk 1 above)
Comment (bmbouter) [reply docs/arch_and_scale/index.rst]: I agree, fixed in next push. ty.

File: docs/arch_and_scale/index.rst (hunk 2 above)
Comment (bmbouter) [reply docs/arch_and_scale/index.rst]: ty, fixed in next push.

File: docs/arch_and_scale/webserver.rst [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  matching ``/pulp/content/`` to the Pulp WSGI application.
+
+Plugin Views
+  Plugins can contribute views anywhere in the url namespace are are not restricted to ``/pulp/``.
+  Refer to your plugin documentation to understand the url needs of any given plugin. Another option
+  is to route ``/``.
+
+Using the urls above you can choose which webservers and how many will serve the different parts of
+the Pulp application. If you want to have a single rule to serve all aspects of the Pulp web
+application routing ``/`` to it will is simple and will always work.
```
Comment (asmacdo) [inline docs/arch_and_scale/webserver.rst]: s/application/application,/ makes this sentence a little more readable.

File: docs/arch_and_scale/webserver.rst [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  through its web application. To serve this content,  have a WSGI compatible webserver route urls
+  matching ``/pulp/content/`` to the Pulp WSGI application.
+
+Plugin Views
+  Plugins can contribute views anywhere in the url namespace are are not restricted to ``/pulp/``.
+  Refer to your plugin documentation to understand the url needs of any given plugin. Another option
+  is to route ``/``.
+
+Using the urls above you can choose which webservers and how many will serve the different parts of
+the Pulp application. If you want to have a single rule to serve all aspects of the Pulp web
```
Comment (asmacdo) [inline docs/arch_and_scale/webserver.rst]: s/aspects/components/ (nitpick, IMO)

File: docs/arch_and_scale/webserver.rst (hunk 3 above)
Comment (bmbouter) [reply docs/arch_and_scale/webserver.rst]: ty. fixed in next push.

File: docs/arch_and_scale/webserver.rst (hunk 4 above)
Comment (bmbouter) [reply docs/arch_and_scale/webserver.rst]: this is good. fixed in next push.

**Comment by dralley** [comment]
@bmbouter need to rebase

**Pull request merged**